### PR TITLE
Support for cross-compiling the swift repl

### PR DESCRIPTION
**Pull request by karwa**
Requires new build-script variable: `LLDB_SWIFTC`, from https://github.com/apple/swift/pull/4574

### CODE REVIEW

**Comment by tfiala** [comment]
@swift-ci Please test Linux platform

**Comment by karwa** [comment]
It will fail to build until the requisite swift build-script variables are given (see linked patch).

**Comment by tfiala** [comment]
Ah I see, okay thanks.

Separately, I have a question into our build system maintainer to figure out why said build did not actually kick off, independently of its expectation to fail.  (We should have gotten a notification of a build going into a queue).

**Comment by karwa** [comment]
@tfiala we have cross-repo testing support now, so it should work

**Comment by tfiala** [comment]
@swift-ci Please test Linux platform

**Comment by karwa** [comment]
Buidbot failed to pull in the linked patch to the swift build script, so it failed.

CC @shahmischal

**Comment by tfiala** [comment]
We're seeing this build error on the Linux CI:
```
[713/872] Building CXX object source/Utility/CMakeFiles/lldbUtility.dir/UriParser.cpp.o
[714/872] Creating export file for liblldb
[715/872] Generating ../../lib/lldb/clang
[716/872] cd /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/tools/repl/swift && /opt/bin/cmake -E remove_directory /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/./repl_swift_module_cache
[717/872] cd /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/tools/repl/swift && /opt/bin/cmake -E make_directory /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/./repl_swift_module_cache
[718/872] Generating ../../../bin/repl_swift
FAILED: cd /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/tools/repl/swift \
&& -resource-dir /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/swift-linux-x86_64/lib/swift \
-module-cache-path /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/./repl_swift_module_cache \
-o /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/Ninja-DebugAssert/lldb-linux-x86_64/./bin/repl_swift \
-Xlinker -rpath -Xlinker "\$ORIGIN/../lib/swift/linux" /home/buildnode/disk2/workspace/swift-lldb-PR-Linux/lldb/tools/repl/swift/main.swift
/bin/sh: 1: -resource-dir: not found
```

**Comment by shahmishal** [comment]
I should have it completed by end of day today.

**Comment by tfiala** [comment]
> Buidbot failed to pull in the linked patch to the swift build script, so it failed.
> CC @shahmischal

Ah sorry, missed that before I reflected up the error.

**Comment by shahmishal** [comment]
https://github.com/apple/swift/pull/4574
@swift-ci Please test Linux platform

**Comment by shahmishal** [comment]
Please test with following PR:
https://github.com/apple/swift/pull/4574

@swift-ci Please test Linux platform

**Comment by tfiala** [comment]
@shahmishal, this doesn't look like it worked.  I'm seeing the same error we were getting this morning.

**Comment by karwa** [comment]
The tests from the swift side of the patch seem to work: https://ci.swift.org/job/swift-PR-Linux/3718/consoleFull

```
(checks out apple/swift from PR 4574, the sister patch)

15:40:03 Fetching upstream changes from git@github.com:apple/swift.git
15:40:03  > git --version # timeout=10
15:40:03  > git -c core.askpass=true fetch --tags --progress git@github.com:apple/swift.git +refs/pull/*:refs/remotes/origin/pr/*
15:40:06  > git rev-parse refs/remotes/origin/pr/4574/merge^{commit} # timeout=10
15:40:06  > git rev-parse refs/remotes/origin/origin/pr/4574/merge^{commit} # timeout=10
15:40:06 Checking out Revision 5cfcc54dea1b7732a085fb38fd1890b36e9eb990 (refs/remotes/origin/pr/4574/merge)
15:40:06  > git config core.sparsecheckout # timeout=10
15:40:06  > git checkout -f 5cfcc54dea1b7732a085fb38fd1890b36e9eb990

... then a bit later it merges this one in to LLDB

15:40:15 + git fetch origin pull/48/merge:ci_pr_48
15:40:16 From https://github.com/apple/swift-lldb
15:40:16  * [new ref]         refs/pull/48/merge -> ci_pr_48
15:40:16 + git fetch --recurse-submodules=yes
15:40:17 From https://github.com/apple/swift-lldb
15:40:17    d5c8597..2412f02  swift-3.0-branch -> origin/swift-3.0-branch
15:40:17 + git clean -fdx
15:40:17 + git submodule foreach --recursive git clean -fdx
15:40:17 + git submodule foreach --recursive git reset --hard HEAD
15:40:18 + git reset --hard HEAD
15:40:18 HEAD is now at a0a6b64 Remove unnecessary include of swift/AST/ExprHandle.h.
15:40:18 + git checkout ci_pr_48
15:40:18 Switched to branch 'ci_pr_48'
15:40:18 + git submodule update --recursive
15:40:18 + git fetch --recurse-submodules=yes
15:40:18 + git clean -fdx
```

It's a little bit hard to follow because there's loads of changes that went in to the same test (and because it was done from the swift side, the swift project is listed in Jenkins' list of changes but the LLDB one isn't). But I think we ended up with everything where it should be.

**Comment by shahmishal** [comment]
@karwa @tfiala Sorry, I am still testing :)

**Comment by tfiala** [comment]
Ah okay thanks!

**Comment by shahmishal** [comment]
Please test with following PR:
apple/swift#4574

@swift-ci Please test Linux platform

**Comment by shahmishal** [comment]
It picked up the right PR, lets see if we see any other issues. 

```
+ git fetch origin pull/4574/merge:ci_pr_4574
From github.com:apple/swift
 * [new ref]         refs/pull/4574/merge -> ci_pr_4574
```

**Comment by tfiala** [comment]
Okay, fingers crossed :-)

**Comment by shahmishal** [comment]
it passed 👍

**Comment by tfiala** [comment]
Awesome, thanks!

**Comment by tfiala** [comment]
@shahmishal I think I put this in prematurely.  We need apple/swift#4574 to go in first, don't we.

**Comment by tfiala** [comment]
I took care of the other check-in.  I restarted the LLDB builders to make sure they picked up the swift change.

**Comment by karwa** [comment]
What happened to this? Did it get lost in a merge? Not in `master`

**Comment by tfiala** [comment]
@llvm-beanz, did we lose this as part of your CMake enhancements, or did this happen as a casualty of a merge?

**Comment by llvm-beanz** [comment]
I can’t find any indication this was ever on trunk. It certainly wasn’t there before my patch merged in.

-Chris
